packages/hardhat/contracts/Ethernauts.sol

```diff
@@ -55,22 +56,31 @@ contract Ethernauts is ERC721Enumerable, Ownable {
         earlyMintPrice = initialEarlyMintPrice;
         couponSigner = initialCouponSigner;
 
-        currentSaleState = SaleState.Paused;
+        _currentSaleState = SaleState.Paused;
     }
 
     // ----------
     // Modifiers
     // ----------
 
     modifier onlyOnState(SaleState definedSaleState) {
-        require(currentSaleState == definedSaleState, "Not allowed in current state");
+        require(getCurrentSaleState() == definedSaleState, "Not allowed in current state");
         _;
     }
 
     // --------------------
     // Public external ABI
     // --------------------
 
+    function getCurrentSaleState() public view returns (SaleState) {
+        if (_currentSaleState < SaleState.Open) {
+            return _currentSaleState;
+        } else {
+            // We need to check if user can mint more tokens
+            return availableToMint() > 0 ? SaleState.Open : SaleState.Completed;
+        }
+    }
+
     function mint() external payable onlyOnState(SaleState.Open) {
```

Comment:
should we update the state automatically? once everything is minted?

Reply:
Yes, it's something we've discussed with @StErMi. We're gonna launch on Optimism, so it won't consume much more gas in each mint calls.

Reply:
@StErMi can you take care of this?